web-transport-quinn/src/recv.rs

```diff
@@ -75,6 +77,16 @@ impl RecvStream {
         }
     }
 
+    /// A stable identifier for this stream.
+    ///
+    /// If this `RecvStream` is part of a bidirectional stream, the `SendStream`
+    /// would have a matching stable id.
+    ///
+    /// This value will remain fixed for the lifetime of the stream.
+    pub fn stable_id(&self) -> usize {
+        &*self.id as *const _ as usize
```

Comment:
I really don't like this approach that Quinn takes. It's technically not unique as memory addresses can be reused (after Drop). I'm not even sure if it's stable, can't memory be moved (Pin vs Unpin nonsense?).

I'd be okay with calling this `quic_id` and returning the underlying QUIC stream ID. We have to very clearly note the restrictions, namely that there may be gaps. It can only be used for creation order and equality.

Reply:
Actually, `quinn::StreamId` doesn't expose any way to retrieve the inner `u64`, so I think we don't really need the note as it'll just be more confusing, i.e. "What do you mean there may be gaps, what are you even talking about?"